Clean up HAVE_PASTEBOARD_DATA_OWNER

emw-apple opened the pull request:
#### 548422a12918d9a17699cc93f3cf813e7fca1205
<pre>
Clean up HAVE_PASTEBOARD_DATA_OWNER
<a href="https://bugs.webkit.org/show_bug.cgi?id=244144">https://bugs.webkit.org/show_bug.cgi?id=244144</a>

Reviewed by Alexey Proskuryakov.

Remove use of this HAVE macro, as the SPI it guards is available on all
iOS family OSes we support. This cleanup simplifies changes needed in
SPI headers to build open source for iOS 16.

As a result of this change, non-iOS-family platforms now declare
WebKit::WebPageProxy::dataOwnerForPasteboard() and
WebKit::PageClient::dataOwnerForPasteboard(). They return
WebCore::DataOwnerType::Undefined.

* Source/WebCore/PAL/pal/spi/ios/UIKitSPI.h: Because
  HAVE_PASTEBOARD_DATA_OWNER was provided via AdditionalFeatureDefines,
  this code was never built or tested in open-source. Making it compile
  requires a bit more SPI.
* Source/WebKit/Platform/spi/ios/UIKitSPI.h: Ditto.
* Tools/TestWebKitAPI/ios/UIKitSPI.h: Ditto.

The rest of the patch is just removing #if blocks, or rewriting them to
be &quot;#if PLATFORM(IOS_FAMILY)&quot;.

* Source/WebCore/platform/ios/PlatformPasteboardIOS.mm:
(WebCore::PlatformPasteboard::performAsDataOwner):
* Source/WebKit/UIProcess/Cocoa/WebPasteboardProxyCocoa.mm:
(WebKit::WebPasteboardProxy::determineDataOwner const):
* Source/WebKit/UIProcess/PageClient.h:
(WebKit::PageClient::dataOwnerForPasteboard const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
* Tools/TestWebKitAPI/Tests/ios/UIPasteboardTests.mm:

Canonical link: <a href="https://commits.webkit.org/254006@main">https://commits.webkit.org/254006@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/05a6690c3a45308845ed1d614584f26018de004e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87516 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31606 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18310 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96743 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/150407 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91489 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29969 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26138 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79635 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91504 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93135 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/24222 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74297 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24044 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79184 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79390 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67062 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27677 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13223 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27630 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14239 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2818 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29320 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37113 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29245 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33517 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/UIProcess/Cocoa/WebPasteboardProxyCocoa.mm:
```diff
@@ -633,7 +633,6 @@
     if (!pageID)
         return DataOwnerType::Undefined;
 
-#if HAVE(PASTEBOARD_DATA_OWNER)
```
Is this iOS only code (i.e. is there an ifdef somewhere outside the visible part)? `HAVE(PASTEBOARD_DATA_OWNER)` appears to be false on macOS. 

Also, where does macCatalyst fall in terms of having it?

emw-apple replied:
Mac Catalyst has it, but let me double check that macOS is safe.

emw-apple replied:
With the latest patch, here's how this works: Any platform may call into `dataOwnerForPasteboard()`, and all except for iOS should return `::Undefined` as specified in the PageClient's default implementation. I've updated the commit message to clarify this.

Arguably this is the wrong thing to do as it imposes an iOS-only concept onto all platforms. @whsieh can probably advise if it's better to simply convert all HAVE(PASTEBOARD_DATA_OWNER) blocks into PLATFORM(IOS), which would match existing behavior but would mean less cleanup.

aproskuryakov commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10469,15 +10469,11 @@ void WebPageProxy::touchBarMenuItemDataRemoved(const TouchBarMenuItemData& touch
 }
 #endif
 
-#if HAVE(PASTEBOARD_DATA_OWNER)
```
Ditto.

aproskuryakov reviewed (approved):
r+ assuming that macOS and macCatalyst situation is understood.

emw-apple committed:
Clean up HAVE_PASTEBOARD_DATA_OWNER
https://bugs.webkit.org/show_bug.cgi?id=244144

Reviewed by Alexey Proskuryakov.

Remove use of this HAVE macro, as the SPI it guards is available on all
iOS family OSes we support. This cleanup simplifies changes needed in
SPI headers to build open source for iOS 16.

As a result of this change, non-iOS-family platforms now declare
WebKit::WebPageProxy::dataOwnerForPasteboard() and
WebKit::PageClient::dataOwnerForPasteboard(). They return
WebCore::DataOwnerType::Undefined.

* Source/WebCore/PAL/pal/spi/ios/UIKitSPI.h: Because
  HAVE_PASTEBOARD_DATA_OWNER was provided via AdditionalFeatureDefines,
  this code was never built or tested in open-source. Making it compile
  requires a bit more SPI.
* Source/WebKit/Platform/spi/ios/UIKitSPI.h: Ditto.
* Tools/TestWebKitAPI/ios/UIKitSPI.h: Ditto.

The rest of the patch is just removing #if blocks, or rewriting them to
be "#if PLATFORM(IOS_FAMILY)".

* Source/WebCore/platform/ios/PlatformPasteboardIOS.mm:
(WebCore::PlatformPasteboard::performAsDataOwner):
* Source/WebKit/UIProcess/Cocoa/WebPasteboardProxyCocoa.mm:
(WebKit::WebPasteboardProxy::determineDataOwner const):
* Source/WebKit/UIProcess/PageClient.h:
(WebKit::PageClient::dataOwnerForPasteboard const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
* Tools/TestWebKitAPI/Tests/ios/UIPasteboardTests.mm:

Canonical link: https://commits.webkit.org/254006@main
